### ISSUE TITLE
The `_raw_md5` field

### ISSUE THREAD

**Issue by glennmatthews** [nautobot/nautobot-plugin-circuit-maintenance#45]
What purpose does the `RawNotification._raw_md5` field serve? I don't see where it's actually being used for anything, and its `unique=True` constraint means that trying to create RawNotifications with the same `raw` body text (e.g from duplicate emails) results in an exception:

```
duplicate key value violates unique constraint "nautobot_circuit_mainten__raw_md5_4f063303_uniq" DETAIL: Key (_raw_md5)=(d41d8cd98f00b204e9800998ecf8427e) already exists.
```

**Comment by chadell** [comment]
From original design, multiple `RawNotification`s can share the same Subject, something such as "Maintenance 123456789" and the unique attribute was the `raw` content, so we can't have multiple `RawNotifications` with exactly the same content, as this is actually a duplicated notification.
Later, I found that using `raw` as the unique key, it was too big for the database, and I created the `_raw_md5` with the only purpose to keep the uniqueness of the notifications.
What is your use-cases to have duplicated notifications?

**Comment by glennmatthews** [comment]
I don't completely understand why it would be a problem to allow multiple RawNotifications with the same `raw` content but potentially different `subject`, `provider`, `sender`, `source`, and `date` values  - perhaps there's something I'm missing here though.

I actually encountered this problem in the context of https://github.com/nautobot/nautobot-plugin-circuit-maintenance/blob/develop/nautobot_circuit_maintenance/handle_notifications/handler.py#L203 - if we failed to parse all of the raw payload(s) in a message, we were setting `raw_payload = b""`, then proceeding to create a `RawNotification` using this payload - so if more than one such message failure occurs, the second and subsequent `RawNotifications` would fail because an empty `raw_payload` had already been saved to the database once before and a second such `RawNotification` would hit the `duplicate key value` error.

I worked around this in https://github.com/nautobot/nautobot-plugin-circuit-maintenance/pull/42/files#diff-1464dbe829f38384b112eb967ada81b11345c7316e4afba03b201975c47b53b6R206 by *not* attempting to create a RawNotification if we were unable to extract any `raw_payload` value at all from a given message.

**Comment by chadell** [comment]
Your are right, it's not a problem, just another choice. 
I agree that we could use something as a `unique` combination of some of the other parameters and skip the`_raw_md5`

**Comment by chadell** [comment]
@glennmatthews while thinking on this I've noticed that the `unique_together` that you proposed has some weak point, because the `date` it's the retrieval time:
```
>>> RawNotification.objects.first().date
datetime.datetime(2021, 8, 11, 5, 13, 58, 22435, tzinfo=<UTC>)
```
Because it is so accurate, it makes all the unique checks to pass, and the same when we retrieve the same notification in different moments. 
Thinking on email notifications, the date could be the date of the received email, but when thinking on external API integrations, it's not sure we will have the "creation" date available.
Skipping the `date` in the `unique_together` is not good enough, because I can imagine that the combination of the other fields can be duplicated when the `subject` of the notification is not actually changing.
Trying to find a better way to skip this `_raw_md5`, but I've not found a better way so far...

**Comment by glennmatthews** [comment]
I think using the send or receive date of the email as the date should be good for now. For future API integrations, I'd expect there would still be some sort of timestamp/datestamp indicating when a notification was added or created; if not, we might have to fall back to using the start time of the actual maintenance as the datestamp?

**Comment by chadell** [comment]
it sounds good ;)

**Comment by glennmatthews** [comment]
This interacts weirdly with the data truncation introduced in #86, as `_raw_md5` is initially calculated based on the full (un-truncated) raw data, which is then truncated. Calling `save()` a second time on the record will result in the `_raw_md5` being recalculated based on the now-truncated raw data. I'd like to see us remove this field sooner than later as in general having a `unique` field that is calculated (or re-calculated) at `save()` time has the potential to throw an IntegrityError which then needs to be handled - I'm seeing this in practice, this is not a theoretical concern. :-)

**Comment by chadell** [comment]
Yes, @Glenn, you are right, we should get rid of it. 
Luckily, now with the new `Date` field, taken from `Stamp` this would be easier.
So, now, I believe that the `unique_together = ("date", "subject", "provider", "source", "sender")` , that you proposed, would work.
do you agree on this?

**Comment by glennmatthews** [comment]
If anything that seems overly cautious to me - I'd expect that (date, subject, provider) should be sufficiently unique, but I'm not opposed to including the others as well if you think there's value.

**Comment by glennmatthews** [comment]
Resolved by #99.
